docs/proposals-done/202203-grpc-query-api.md

```diff
@@ -0,0 +1,108 @@
+## Introduce a Query gRPC API
+
+* **Owners:**
+    * `@fpetkovski` 
+    * `@mzardab` 
+
+> TL;DR: Introducing a new gRPC API for `/query` and `/query_range`
+
+## Why
+We want to be able to distinguish between gRPC Store APIs and other Queriers in the query path. Currently, Thanos Query implements the gRPC Store API and the root Querier does not distinguish between Store targets and other Queriers that are capable of processing a PromQL expression before returning the result. The new gRPC Query API will allow a querier to fan out query execution, in addition to Store API selects.
+
+This is useful for a few reasons: 
+
+* When Queriers register disjoint Store targets, they should be able to deduplicate series and then execute the query without concerns of duplicate data from other queriers. This new API would enable users to effectively partition by Querier, and avoid shipping raw series back from each disjointed Querier to the root Querier.
+* If Queriers register conjoint Store targets, users would be able to express a query sharding strategy between Queriers to more effectively distribute query load amongst a fleet of homogenous Queriers. 
```

Comment:
What `conjoint` might mean in this context?

Reply:
It means overlapping/interleaved series. Updated the sentence